R/with.r

```diff
@@ -128,9 +143,13 @@ set_libpaths <- function(paths) {
   invisible(old)
 }
 
+reset_libpaths <- function(paths) {
```

Comment:
@krlmrlr I started moving the `with_` functions a few months ago, https://github.com/jimhester/robustr, made a few improvements (including more useful function definitions when printing), but I haven't had time to give it a final polish and convert devtools to use it.